src/vdom.rs

```diff
@@ -72,45 +103,50 @@ pub(crate) fn build(host: &Host, input: VDom<()>) -> VDomNode<Vec<VDom<Node>>, N
                 })
                 .collect();
             // Return the machine
-            VDomNode::Elem {
-                name,
-                attrs: attrs_new,
-                children,
-                state: node,
-            }
+            VDomNode::new_elem(elem.name, attrs_new, children, node)
         }
     }
 }
 
 impl VDomNode<Vec<VDom<Node>>, Node, Attrs> {
+    pub fn new_text(text: String, state: Node) -> Self {
+        VDomNode::Text(Text::new(text, state))
+    }
+
+    pub fn new_elem(name: String, attrs: Attrs, children: Vec<VDom<Node>>, state: Node) -> Self {
+        VDomNode::Elem(Elem::new(name, attrs, children, state))
+    }
     pub(crate) fn node(&self) -> &Node {
         match self {
-            VDomNode::Text { state, .. } => state,
-            VDomNode::Elem { state, .. } => state,
+            VDomNode::Text(text) => &text.state,
+            VDomNode::Elem(elem) => &elem.state,
         }
     }
 }
 
 impl VDom<Node> {
+    pub fn new(vdom: VDomNode<Vec<VDom<Node>>, Node, Attrs>) -> Self {
```

Comment:
VDom::new should also be defined without depending on [Node] as the state

Reply:
Changing `Node` to `State` leads to install function not working with `host.install(...)` because `install` function consumes parameters of type `Node`.
```rs
    pub(crate) fn install(&self, node: &Node) {
        self.body.append_child(&node.element);
    }
```
for the following impls
```rs
impl<State> VDom<State> {
    pub fn new(vdom: VDomNode<Vec<VDom<State>>, State, Attrs>) -> Self {
        VDom { vdom }
    }

    pub fn install(&self, host: &Host) {
        host.install(self.vdom.node());
    }
    ....
    ....
 ```

Reply:
Yeah install works only for `Node`. I suggest we put `install` and `new` in separate impl blocks